AppLovin MAX Demo App - Swift/AppLovin MAX Demo App - Swift/Base Classes/MAAdViewSwiftUIWrapper.swift

```diff
@@ -0,0 +1,161 @@
+    {
+        let adView = MAAdView(adUnitIdentifier: adUnitIdentifier, adFormat: adFormat, sdk: sdk)
+        
+        adView.delegate = context.coordinator
+        adView.requestDelegate = context.coordinator
+        adView.revenueDelegate = context.coordinator
+        
+        // Set background or background color for MRECs to be fully functional
+        adView.backgroundColor = .black
+        
+        // Load the first ad
+        adView.loadAd()
+        
+        return adView
+    }
+    
+    func updateUIView(_ uiView: MAAdView, context: Context) {}
+    
+    func makeCoordinator() -> Coordinator
+    {
+        Coordinator(parent: self)
+    }
+}
+
+@available(iOS 13.0, *)
+extension MAAdViewSwiftUIWrapper
+{
+    class Coordinator: NSObject, MAAdViewAdDelegate, MAAdRequestDelegate, MAAdRevenueDelegate
+    {
+        private let parent: MAAdViewSwiftUIWrapper
+        
+        init(parent: MAAdViewSwiftUIWrapper)
+        {
+            self.parent = parent
+        }
+        
+        func didStartAdRequest(forAdUnitIdentifier adUnitIdentifier: String)
+        {
+            parent.didStartAdRequest?(adUnitIdentifier)
+        }
+        
+        func didLoad(_ ad: MAAd)
+        {
+            parent.didLoad?(ad)
+        }
+        
+        func didFailToLoadAd(forAdUnitIdentifier adUnitIdentifier: String, withError error: MAError)
+        {
+            parent.didFailToLoadAd?(adUnitIdentifier, error)
+        }
+        
+        func didDisplay(_ ad: MAAd)
+        {
+            parent.didDisplay?(ad)
+        }
+        
+        func didFail(toDisplay ad: MAAd, withError error: MAError)
+        {
+            parent.didFailToDisplayAd?(ad, error)
+        }
+        
+        func didClick(_ ad: MAAd)
+        {
+            parent.didClick?(ad)
+        }
+        
+        func didExpand(_ ad: MAAd)
+        {
+            parent.didExpand?(ad)
+        }
+        
+        func didCollapse(_ ad: MAAd)
+        {
+            parent.didCollapse?(ad)
+        }
+        
+        func didHide(_ ad: MAAd)
+        {
+            parent.didHide?(ad)
+        }
+        
+        func didPayRevenue(for ad: MAAd)
+        {
+            parent.didPayRevenue?(ad)
+        }
+    }
+}
+
+@available(iOS 13.0, *)
+extension MAAdViewSwiftUIWrapper
+{
+    func deviceSpecificFrame() -> some View
+    {
+        modifier(MAAdViewFrame(adFormat: adFormat))
+    }
+}
+
+@available(iOS 13.0.0, *)
+struct MAAdViewFrame: ViewModifier
+{
+    let adFormat: MAAdFormat
+    
+    func body(content: Content) -> some View
+    {
+        if ( adFormat == .banner || adFormat == .leader )
+        {
+            // Stretch to the width of the screen for banners to be fully functional
+            // Banner height on iPhone and iPad is 50 and 90, respectively
+            content
+                .frame(width: UIScreen.main.bounds.width,
+                       height: (UIDevice.current.userInterfaceIdiom == .pad) ? 90 : 50)
```

Comment:
```suggestion
                .frame(height: (UIDevice.current.userInterfaceIdiom == .pad) ? 90 : 50)
```
In SwiftUI, views expand to the maximum size possible by default. So we can just drop the width argument here, but leave the comment as an explanation.